src/client.rs

```diff
@@ -223,85 +223,109 @@ mod tests {
+            Test {
+                test_description: "test_get_fails_if_verification_fails_with_verification_error",
+                client: Client::create_with_server_collection(&mock_server.url(""), "url-classifier-skip-urls", Some(Box::new(VerifierWithVerificatonError{}))),
+                changeset_response_body:  r#"{
+                    "metadata": {},
+                    "changes": [],
+                    "timestamp": 0
+                }"#,
+                expected_result: Err(ClientError::VerificationError{name: "verification error".to_owned()})
             },
-            Err(error) => panic!("invalid response : {:?}", error)
-        };
-    }
+            Test {
+                test_description: "test_get_passes_if_verification_passes",
+                client: Client::create_with_server_collection(&mock_server.url(""), "url-classifier-skip-urls", Some(Box::new(VerifierWithNoError{}))),
+                changeset_response_body: r#"{
+                    "metadata": {
+                        "data": "test"
+                    },
+                    "changes": [{
+                        "id": 1,
+                        "last_modified": 100
+                    }],
+                    "timestamp": 0
+                }"#,
+                expected_result: Ok(vec![json!({
+                        "id": 1,
+                        "last_modified": 100
+                    })
+                ])
+            },
+            Test {
+                test_description: "test_get_fails_if_verification_fails_with_invalid_signature_error",
+                client: Client::create_with_server_collection(&mock_server.url(""), "url-classifier-skip-urls", Some(Box::new(VerifierWithInvalidSignatureError{}))),
+                changeset_response_body: r#"{
+                    "metadata": {},
+                    "changes": [],
+                    "timestamp": 0
+                }"#,
+                expected_result: Err(ClientError::Error{name: "invalid signature error".to_owned()})
+            }
+        ];
 
-    #[tokio::test]
-    #[with_mock_server]
-    async fn test_get_fails_if_verification_fails_with_invalid_signature_error() {
-        let get_changeset_mock = mock(GET, "/buckets/main/collections/url-classifier-skip-urls/changeset")
-        .expect_query_param("_expected", "0")
-        .return_status(200)
-        .return_header("Content-Type", "application/json")
-        .return_body(
+        for test in &tests {
+
+            debug!("test name: {}", test.test_description);
+
+            let mut get_latest_change_mock = Mock::new()
+            .expect_method(GET)
+            .expect_path("/buckets/monitor/collections/changes/records")
+            .expect_query_param("bucket", "main")
+            .expect_query_param("collection", "url-classifier-skip-urls")
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_header("etag", "12345")
```

Comment:
Is this used somewhere?

Reply:
to get the timestamp from the response

Reply:
and what do we do with it? I don't think it's necessary

Reply:
its used for debugging but nothing other than that - I can remove it